frontend/dockerfile/instructions/parse_test.go

```diff
@@ -140,6 +141,36 @@ func TestParseOptInterval(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestCommentsDetection(t *testing.T) {
+	dt := `# foo sets foo
+ARG foo=bar
+
+# base defines first stage
+FROM busybox AS base
+# this is irrelevant
+ARG foo
+# bar defines bar
+ARG bar
```

Comment:
Does the code support (and if so), could we add a test for a comment that documents an `ARG` that specifies multiple args? e.g.;

```dockerfile
# bar defines bar
# foo defines foo
ARG bar foo=qux
```

In the above, I assume order is important(?) so this wouldn't match anything;

```dockerfile
# foo defines foo
# bar defines bar
ARG bar foo=qux
```

Reply:
Also wondering (as a follow-up if we think it's a good idea) to allow documenting `VOLUME`;

```dockerfile
# /app/data holds the application's persistent data
VOLUME /app/data

# /tmp/appcache is an ephemeral storage for the application's runtime data
VOLUME /tmp/appcache
```

And perhaps `ENV` and `LABEL` as well? Documenting what the purpose is of specific `ENV` and `LABEL`s

```dockerfile
# VERSION holds the version of foo that is installed
ENV VERSION=v1.0.0-beta.2

# com.foo.git-commit holds the git reference that was used to build this image
LABEL com.foo.git-commit=deadbeef
```

Reply:
For the volume/env/label you can of course add comments like this. But there isn't a use case to parse them out as they are not useful without a Dockerfile itself. Current locations where they are parsed out are positions that can be configured by name during a build.

Reply:
added to multi-arg test case